toolkit/scripts/pkggen.mk

```diff
@@ -89,13 +89,29 @@ $(specs_file): $(chroot_worker) $(BUILD_SPECS_DIR) $(build_specs) $(build_spec_d
 		$(if $(TARGET_ARCH),--target-arch="$(TARGET_ARCH)") \
 	        --output $@
 
+ifeq ($(RESOLVE_CYCLES_FROM_UPSTREAM),y)
+   ifeq ($(DISABLE_UPSTREAM_REPOS),y)
+      $(error RESOLVE_CYCLES_FROM_UPSTREAM requires upstream repos to be enabled. Please set DISABLE_UPSTREAM_REPOS=n)
+   endif
+endif
+
 # Convert the dependency information in the json file into a graph structure
 # We require all the toolchain RPMs to be available here to help resolve unfixable cyclic dependencies
-$(graph_file): $(specs_file) $(go-grapher) $(toolchain_rpms)
+$(graph_file): $(specs_file) $(go-grapher) $(toolchain_rpms) $(rpm_cache_files) $(TOOLCHAIN_MANIFEST) $(chroot_worker) $(pkggen_local_repo) $(depend_REPO_LIST)
```

Comment:
>pkggen_local_repo [](http://example.com/codeflow?start=121&length=17)

I like it that you added `$(pkggen_local_repo)` as a dependency. Let's also add `$(graphpkgfetcher_cloned_repo)`.

Reply:
done